text/0000-export-to-oci.md

```diff
@@ -0,0 +1,246 @@
+    │       ├── 6977801a3b8ac0c4a76ea22fa4dc0541fe8fc6ba964883ee0a21b5736f8acee9
+    │       ├── 6b534265b45a4c09a5e05dde5ccf2450ea505481658a1c9ae9e0aae32362e941 -> ../../../cnbs/sample-stack-run/blobs/sha256/6b534265b45a4c09a5e05dde5ccf2450ea505481658a1c9ae9e0aae32362e941
+    │       ├── 7c83d7d24ac43dbdd419d15abb849b6d155d0bc361eca0908a8ae5aefcd55557
+    │       └── addd2357f3a0175410ab8f9303e747cc72af9aaeb0e7402d3fd3f144adb29db5
+    ├── index.json
+    └── oci-layout
+```
+
+Also, we are proposing to use symbolic links to reference the blobs from the `run-image` this will help to save some hard drive space for the end user.
+
+### Using -oci flag in combination with --daemon or --publish flags
+
+Any combination of using multiple sources or sinks in the Lifecycle invocation of phases should throw an error to the user. For example:
+
+```=shell
+> /cnb/lifecycle/exporter -oci -daemon -run-image cnbs/sample-stack-run:bionic my-app-image
+
+ERROR: exporting to multiples target is not allowed
+```
+
+# How it Works
+[how-it-works]: #how-it-works
+
+The lifecycle phases affected by this new behavior are:
+ - [Analyze](https://buildpacks.io/docs/concepts/components/lifecycle/analyze/)
+ - [Export](https://buildpacks.io/docs/concepts/components/lifecycle/export/)  
+ - [Create](https://buildpacks.io/docs/concepts/components/lifecycle/create/)
+
+At high level view the proposed solution can be summarize with the following system landscape diagram from the C4 model
+
+![](https://i.imgur.com/y972lTD.png)
+
+Notice that we are relying on the OCI format Specification to expose the data for `Platforms`
+
+The following new input are proposed to be added to these phases
+
+ | Input             | Environment Variable  | Default Value            | Description
+ |-------------------|-----------------------|--------------------------|----------------------
+ | `<oci>`      |  `CNB_USE_OCI` | false | Analyze or Export image from/to OCI layout format on disk |
+ | `<oci-dir>` | `CNB_OCI_PATH` | /oci | Path to oci directory where the images are saved |
+
+- WHEN `the new flag -oci or the default environment variable CNB_USE_OCI are set to true` during the phase invocation THEN the feature will be enabled.
+The image look up will be done following these rules:
+  - WHEN `the image points to a tag reference`
+    - Lifecycle will load the image from disk in [OCI Image Layout](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) format at `<oci-dir>/<registry>/<repo>/<tag>`
+  - WHEN `the image points to a digest reference`
+    - Lifecycle will load the image from disk in [OCI Image Layout](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) format at `<oci-dir>/<registry>/<repo>/<digest>`
+  - A part from the look up, the logic for each phase should remain the same
+
+# Migration
+[migration]: #migration
+
+- No breaking changes were identified
```

Comment:
Ah, maybe my suggestion [here](https://github.com/buildpacks/rfcs/pull/203/files#r891354463) was wrong - does this RFC propose removing the daemon? I see daemon removal as one of the motivations. If we are going to propose removing the daemon, there are definitely breaking changes! Are we deferring that discussion to a later RFC? To me, export-to-oci and daemon-removal feel very intertwined.